OktaIdxExample/OktaIdxExampleUITests/SelfServiceRegistrationScenarioTests.swift

```diff
@@ -276,23 +277,31 @@ class SelfServiceRegistrationScenarioTests: XCTestCase {
         let codeExpectation = expectation(description: "Email code received.")
         var emailCode: String?
         
-        a18nProfile.message { (message: A18NProfile.EmailMessage?, error) in
-            emailCode = message?.content
+        // Wait the code received by email box
+        Thread.sleep(forTimeInterval: TimeInterval.minimal)
+        
+        let emailReceiver = EmailCodeReceiver(profile: a18nProfile)
+        emailReceiver.receiveCode { code in
+            emailCode = code
             
             codeExpectation.fulfill()
         }
         
-        wait(for: [codeExpectation], timeout: 15)
+        wait(for: [codeExpectation], timeout: .regular)
 
         emailCodePage.passcodeField.typeText(try XCTUnwrap(emailCode))
         
         continueButton.tap()
+        
+        // Sometimes tests are very quick. And there's a strange bug after Continue button pressed.
+        // UI is updated faster than the events delivered
+        Thread.sleep(forTimeInterval: 2)
```

Comment:
Usually if you have an arbitrary sleep, it means you need to wait for some condition before proceeding. Perhaps waiting until a label goes away; perhaps the title of the current remediation option or something?  That way control doesn't leave this function until the UI has fully dismissed the controller.

Reply:
I left a comment to return back to it. 
I am aware about the waiting elements, but the issue is another. 
The issue become when the Continue button pressed, picker appears, test selects SMS value, and then the page is reloaded. Obviously the reason is not in waiting the element. Moreover, I cannot reproduce that in debug mode (step by step).

Reply:
I've had some intermittent problems with the picker view, even when using the app directly. So I'll likely refactor this later to use a different UX pattern, which might simplify matters.